pyro/params/param_store.py

```diff
@@ -96,6 +119,8 @@ def __getitem__(self, name):
         Get the constrained value of a named parameter.
         """
         unconstrained_value = self._params[name]
+        if self._no_grad:
+            unconstrained_value = self._fetch_no_grad(unconstrained_value)
```

Comment:
What is the advantage of mutating `unconstrained_value.requires_grad` rather than `.detach()`ing?

Reply:
I was planning to use `.detach` initially, but it still gives the problem of `RuntimeError: Cannot insert a Tensor that requires grad as a constant. Consider making it a parameter or input, or detaching the gradient`. My guess is that `detach`ing creates a new wrapper but the old tensor still exists in the param store and still has requires_grad set to True. As such, `jit.trace` complains about tensor constants that require grad. Whereas if we set `.requires_grad_`, it sets the flag on the existing tensor itself and the tracer is happy when all constant tensors don't require grad.